### PR TITLE
Travis CI support

### PR DESCRIPTION
**Pull request by ayurchen**
- fixes compilation with clang, including a few changes in the code which this version of clang didn't like
- does away with galerautils.hpp header to minimize includes
- other compilation cleanups
- switches GCache malloc interface from ssize_t to int since it should be identical on both 64 and 32-bit systems

The main results of these changes:
- adds support for compiling by clang, which found a number of issues, like unused class members
- speeds up compilation by GCC by a factor of 1.3
  https://docs.google.com/spreadsheets/d/18IiS-EpIf4x4eQavabXX9lDef7xQglQeT7yR5bhvg78/edit?usp=sharing

### CODE REVIEW

**Comment by ayurchen** [comment]
Philip, assigning this to you. Merge at your discretion.

File: galerautils/src/gu_dbug.c [hunk 1]
```diff
@@ -1532,7 +1532,7 @@ GU_DBUGOpenFile(const char *name, int append)
 
     if (name != NULL) {
 	strcpy(_gu_db_stack->name, name);
-	if (strcmp(name, "-") == 0) {
+	if (strlen(name) == 1 && name[0] == '=') {
```
Comment (temeo) [inline galerautils/src/gu_dbug.c]: What is the reason for this change? Some micro optimization in debug code?

Besides, string length check is unnecessary and there is a typo '-' changes to '='.

Comment (temeo) [inline galerautils/src/gu_dbug.c]: Oops, sorry. String length check is necessary but the typo is still there.

File: galerautils/src/gu_mutex.c [hunk 2]
```diff
@@ -21,6 +21,7 @@ gu_mutex_init = { .target_mutex      = PTHREAD_MUTEX_INITIALIZER,
                   .file              = __FILE__,
                   .line              = __LINE__
 };
+#endif
```
Comment (temeo) [inline galerautils/src/gu_mutex.c:24]: Would it be better to delete the definition altogether if it is not used anywhere?

File: gcs/src/gcs_act_proto.cpp [hunk 3]
```diff
@@ -25,15 +25,15 @@ AT - action type
...
+// static const size_t PROTO_ACT_ID_OFFSET   = 0;
+// static const size_t PROTO_ACT_SIZE_OFFSET = 8;
+// static const size_t PROTO_FRAG_NO_OFFSET  = 12;
 
-static const gcs_seqno_t   PROTO_ACT_ID_MAX   = 0x00FFFFFFFFFFFFLL;
-static const unsigned int  PROTO_FRAG_NO_MAX  = 0xFFFFFFFF;
-static const unsigned char PROTO_AT_MAX       = 0xFF;
+// static const gcs_seqno_t   PROTO_ACT_ID_MAX   = 0x00FFFFFFFFFFFFLL;
+// static const unsigned int  PROTO_FRAG_NO_MAX  = 0xFFFFFFFF;
+// static const unsigned char PROTO_AT_MAX       = 0xFF;
```
Comment (temeo) [inline gcs/src/gcs_act_proto.cpp:36]: Again, why not to remove these definitions altogether if they are not used?

**Comment by temeo** [comment]
> switches GCache malloc interface from ssize_t to int since it should be identical on both 64 and 32-bit systems

Why does internal interface need to be identical in 64 and 32 bit systems?

File: galerautils/src/gu_dbug.c (hunk 1 above)
Comment (ayurchen) [inline galerautils/src/gu_dbug.c]: Blast, there is indeed a typo, fix coming.
However the change was needed because clang failed to compile it. It (or libc) expanded strcmp() into some HUGE and barely parseable script that resulted in array boundary violation. I knew I needed to comment on that, but commenting code for a code **change** felt strange... The code itself is rather clear though.

File: gcs/src/gcs_act_proto.cpp (hunk 3 above)
Comment (ayurchen) [inline gcs/src/gcs_act_proto.cpp:36]: To all above - these are not used ATM, but still serve as clarification to the code (to me)

**Comment by ayurchen** [comment]
> Why does internal interface need to be identical in 64 and 32 bit systems?

Ok, maybe it was not worded in the best way. This change was also prompted by clang complains (about shadowing virtual methods due to argument signedness) and along with the fix to that I decided to switch from size_t to int since this is what is the overall limit on a writeset size (which should be the same on 64 and 32 bit systems). Of course (s)size_t on 64 bit systems covers it, but here we give compiler a chance to do something smart. Plus maybe some static checks about narrowing the variable range.

**Comment by temeo** [comment]
Looks ok to merge now.

**Comment by philip-galera** [comment]
Thanks, merging.
